pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -107,6 +115,23 @@ public ProducerImpl(PulsarClientImpl client, String topic, String producerName,
         this.compressor = CompressionCodecProvider
                 .getCompressionCodec(convertCompressionType(conf.getCompressionType()));
 
+        if (conf.isEncryptionEnabled()) {
+            this.msgCrypto = new MessageCrypto(true);
+
+            // Regenerate data key cipher at fixed interval
+            keyGenExecutor = Executors.newSingleThreadScheduledExecutor();
+            keyGenExecutor.scheduleWithFixedDelay(() -> {
+                try {
+                    msgCrypto.addPublicKeyCipher(conf.getEncryptionKeys(), conf.getCryptoKeyReader());
+                } catch (CryptoException e) {
+                    if (!producerCreatedFuture.isDone()) {
```

Comment:
Should we log `warn` message, else there is no way to figure out if key stopped getting generated?

Reply:
Added warn msg.